### PR TITLE
adding support for ResIOT gateways

### PR DESCRIPTION
**Pull request by StevenResiot**

### CODE REVIEW

File: package/resiot/control/control [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Package: helium_gateway
+Version: ${CARGO_MAKE_CRATE_VERSION}
+Architecture: armv7ahf-vfp
```
Comment (madninja) [inline package/resiot/control/control:3]: This should probably be ${CARGO_MAKE_PROFILE} since it now appears to point at slightly different architecture?

Comment (StevenResiot) [reply package/resiot/control/control:3]: Had to pick one from our opkg architecture list as the replaced variable does not appear in the default list. I assumed editing it here was faster than reflashing all the gateways or manually editing opkg configuration settings on the gateways before installing

Comment (madninja) [reply package/resiot/control/control:3]: ugh.. you'd think that the architecture would match the build arch target.. oh well. if it works for ya

**Comment by madninja** [comment]
If you could rebase your PR on main to fix up the ci flow and readme?

File: Makefile.toml [hunk 2]
```diff
@@ -101,7 +106,7 @@ script = '''
 	cd ${CARGO_MAKE_WORKING_DIRECTORY}
 
 	cd ${PKG_BUILD_DIR}
-	if [ "true" = "${IPK_USE_AR}-unset}" ]; then
+	if [ "true" = "${IPK_USE_AR}" ]; then
 		rm ../${PKG_NAME}
```
Comment (madninja) [inline Makefile.toml:110]:
```suggestion
		rm -f ../${PKG_NAME}
```

File: package/resiot/control/prerm [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+/etc/init.d/helium_gateway stop
```
Comment (madninja) [inline package/resiot/control/prerm:3]: remove this line.. auto upgrades when the system tries to stop during the upgrade

**Review by madninja — CHANGES_REQUESTED** [review]

**Review by madninja — APPROVED** [review]

**Pull request merged**